android/src/main/java/nl/fhict/happynews/android/activity/SourcesSettingsActivity.java

```diff
@@ -28,16 +32,26 @@ protected void onCreate(Bundle savedInstanceState) {
         getSupportActionBar().setDisplayHomeAsUpEnabled(true);
         setTitle(R.string.title_settings_sources);
 
-        preferences = getApplicationContext().getSharedPreferences(
-            getString(R.string.preference_file_key), Context.MODE_PRIVATE);
-
         sourcesListView = (ListView) findViewById(R.id.sourcesListView);
-        sources = createSourcesObjects();
+        sources = SourceController.getInstance().getSources(getApplicationContext());
+        Collections.sort((List<SourceSetting>)sources);
         refreshList();
         sourcesAdapter.setParentActivity(this);
     }
 
     /**
+     * Back button implementation.
+     *
+     * @param item menuItem
+     * @return boolean start activity
+     */
+    public boolean onOptionsItemSelected(MenuItem item) {
```

Comment:
This back button implementation is a bit buggy, I have already removed it from the other activities, the default works just fine.